### PR TITLE
Recover if crashing between posting root msg and adding self to group

### PR DESCRIPTION
**Pull request by Powersource**
for https://github.com/ssbc/ssb-tribes2/issues/40

### CODE REVIEW

**Comment by Powersource** [comment]
aah a test is failing

File: index.js [hunk 1]
```diff
@@ -154,59 +156,137 @@ module.exports = {
...
+            subfeed: groupFeed.keys,
+          }
+
+          ssb.box2.addGroupInfo(data.id, {
+            key: data.secret,
+            root: data.root,
+          })
+
+          // Adding myself for recovery reasons
+          addMembers(data.id, [myRoot.id], {}, (err) => {
```
Comment (Powersource) [inline index.js:280]: do we need to reindexencrypted after adding ourselves? didn't seem to help but here's a diff

```diff
diff --git a/index.js b/index.js
index 385bb86..fff02cb 100644
--- a/index.js
+++ b/index.js
@@ -281,7 +282,11 @@ module.exports = {
             // prettier-ignore
             if (err) return cb(clarify(err, 'Failed to add myself to the group when creating a group'))
 
-            return cb(null, data)
+            ssb.db.reindexEncrypted((err) => {
+              // prettier-ignore
+              if (err) return cb(clarify(err, "failed to reindexencrypted after adding self to newly created group"))
+              return cb(null, data)
+            })
           })
         }
       )

```

**Comment by Powersource** [comment]
@staltz @mixmix fixed tests!

**Review by staltz — APPROVED** [review]
Looks good to merge. The implementation is getting trickier, and maybe eventually it'll make sense to create easy APIs to query groups and our group feeds, but no need to anticipate that right now

**Pull request merged**